source/common/io/io_uring_impl.cc

```diff
@@ -5,6 +5,18 @@
 namespace Envoy {
 namespace Io {
 
+bool isIoUringSupported() {
+  struct io_uring_params p {};
+  struct io_uring ring;
+
+  bool is_supported = io_uring_queue_init_params(2, &ring, &p) == 0;
```

Comment:
Read the  [code](https://github.com/axboe/liburing/blob/8695f66bccda2170d8af2f99d3c1b8c6ee8bf602/src/setup.c#L132), and I agree that 0 is the valid result of the function.

Should we use the setup syscall directly or some alternative way to validate io_uring support?

Reply:
The syscall number for io_uring_setup may differ for different archs and libc may lack correct wrappers. I'd rather let liburing handle these subtleties.